zebra-rpc/src/methods.rs

```diff
@@ -526,6 +549,52 @@ where
         }
         .boxed()
     }
+
+    fn get_address_tx_ids(
+        &self,
+        addresses: Vec<String>,
+        start: u32,
+        end: u32,
+    ) -> BoxFuture<Result<Vec<String>>> {
+        let mut state = self.state.clone();
+        let mut response_transactions = vec![];
+        let start = Height(start);
+        let end = Height(end);
+
+        async move {
+            if start > end {
+                return Err(Error::invalid_params(
+                    "End height should be at least start height",
+                ));
+            }
+
+            for address in addresses {
```

Comment:
Performance: Can you pass a list of addresses in a single state request, rather than doing a request for each address?

Some wallets can have hundreds of addresses.

Reply:
Understood, changed in https://github.com/ZcashFoundation/zebra/pull/4062/commits/0948c7ee5af61c9c20900255d67f5ec5a96ec581